models/issue_reaction.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2017 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	//"fmt"
+	//"fmt"
+	"time"
+
+	"github.com/go-xorm/builder"
+	"github.com/go-xorm/xorm"
+)
+
+// Reaction represents a reactions on issues and comments.
+type Reaction struct {
+	ID          int64     `xorm:"pk autoincr"`
+	Type        string    `xorm:"INDEX UNIQUE(s) NOT NULL"`
+	IssueID     int64     `xorm:"INDEX UNIQUE(s) NOT NULL"`
+	CommentID   int64     `xorm:"INDEX UNIQUE(s)"`
```

Comment:
Foreign key please? 😄

Reply:
@bkcsoft IIRC, xorm does not support foreign keys :cry:

Reply:
And what also I don't quite like it inserts 0 values for IDs where it would normally should be null

Reply:
Why would it _ever_ insert `0`/`null` ? It would throw `foreign key constrain` and that would be that 😛